sc/social/like/utils.py

```diff
@@ -112,3 +117,34 @@ def get_valid_objects(brains):
             logger.warn(msg.format(b.getPath()))
             continue
         yield obj
+
+
+def get_image_fallback(site=None):
+    from sc.social.like.interfaces import ISocialLikeSettings
+    if site is None:
+        site = getSite()
+    registry = getUtility(IRegistry)
+    settings = registry.forInterface(ISocialLikeSettings, check=False)
+    site_url = site.absolute_url()
+
+    if getattr(settings, 'image_fallback', False):
+        filename, data = b64decode_file(settings.image_fallback)
+        return '{0}/@@sociallike-image-fallback/{1}'.format(
+            site_url, filename)
+    else:
+        return '{0}/logo.png'.format(site_url)
+
+
+# def validate_image_settings(value):
+#     """Check image fallback be in formats mime type, dimensions and size."""
+#
+#     if not value:
+#         return True
+#
+#     filename, data = b64decode_file(value)
+#     image = NamedBlobImage(data=data, filename=filename)
+#     msg = validate_image_social(image)
```

Comment:
the validator no longer returns the message: it will rise `ValueError` with the error message; you may intercept and raise `Invalid(e.message)` instead.

Reply:
I did not finish waiting for # 136